default.nix

```diff
@@ -0,0 +1,34 @@
+{ # Fetch the latest haskell.nix and import its default.nix
+  haskellNix ? import (builtins.fetchTarball "https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz") {}
+
+# haskell.nix provides access to the nixpkgs pins which are used by our CI,
+# hence you will be more likely to get cache hits when using these.
+# But you can also just use your own, e.g. '<nixpkgs>'.
+, nixpkgsSrc ? haskellNix.sources.nixpkgs-2003
+
+# haskell.nix provides some arguments to be passed to nixpkgs, including some
+# patches and also the haskell.nix functionality itself as an overlay.
+, nixpkgsArgs ? haskellNix.nixpkgsArgs
+, compiler-nix-name ? "ghc884"
+}: 
+let
+  # Define custom nixpkgs overlay
+  overlays = haskellNix.overlays ++ [
+    (self: super: {
+        sdl2 = super.haskellPackages.sdl2;
+    })
+  ];
```

Comment:
This doesn't look right. I guess Haskell.nix tries to build the Haskell SDL2 library and complains about the `sdl2` `pkg-config` dependency? If so, we need to send a PR to Haskell.nix to solve this. Happy to look into this if you show me the error you get if you remove this overlay.

Reply:
Yes: 

```
error: The Nixpkgs package set does not contain the package: sdl2 (system dependency).
You may need to augment the system package mapping in haskell.nix so that it can be found.
```
per [`haskell.nixlib/pkgconf-nixpkgs-map.nix`](https://github.com/input-output-hk/haskell.nix/blob/36a85c8b3483b5a8fbca5ccabd0f71b08b2d9623/lib/pkgconf-nixpkgs-map.nix), `sdl2` is mapped to `SDL2` not `haskellPackages.sdl2`

Reply:
Hmm, but that's correct - this is about the C library which _is_ `pkgs.SDL2`. Can I see the full build log up to failure? Is it failing to build `dear-imgui` or the `sdl2` dependency?

Reply:
```
% nix-build -A dear-imgui.components.library --show-trace
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
trace: No index state specified for dear-imgui, using the latest index state that we know about (2021-01-25T00:00:00Z)!
error: while evaluating the attribute '__impureHostDeps' of the derivation 'dear-imgui-lib-dear-imgui-1.0.0' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/pkgs/stdenv/generic/make-derivation.nix:192:11:
while evaluating anonymous function at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/pkgs/stdenv/generic/make-derivation.nix:175:36, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/pkgs/stdenv/generic/make-derivation.nix:175:21:
while evaluating 'chooseDevOutputs' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/attrsets.nix:475:22, called from undefined position:
while evaluating the attribute 'libs' at undefined position:
while evaluating 'g' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/attrsets.nix:276:19, called from undefined position:
while evaluating anonymous function at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:98:72, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/attrsets.nix:279:20:
while evaluating the attribute 'value' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:465:9:
while evaluating the option `packages.dear-imgui.components.library.libs':
while evaluating the attribute 'mergedValue' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:497:5:
while evaluating 'merge' at /nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/plan.nix:39:18, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:499:59:
while evaluating anonymous function at /nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/plan.nix:40:33, called from undefined position:
while evaluating anonymous function at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/lists.nix:116:29, called from undefined position:
while evaluating anonymous function at /nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/plan.nix:42:21, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/lists.nix:116:32:
while evaluating the attribute 'optionalValue' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:509:5:
while evaluating the attribute 'values' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:491:9:
while evaluating the attribute 'values' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:590:7:
while evaluating anonymous function at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:477:28, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:477:17:
while evaluating definitions from `/nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/plan.nix':
while evaluating 'dischargeProperties' at /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:549:25, called from /nix/store/9wkvnaxphqx96pq7ga6rbf1klxfp9xvq-source/lib/modules.nix:478:137:
while evaluating the attribute 'value' at /nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/plan.nix:46:38:
while evaluating 'sysDepError' at /nix/store/1f6fbxz6x90k90qfkwjmgrpivdz75w3l-source/modules/error-handler.nix:13:19, called from /nix/store/7mcaxsq6sk28fi53gwdxygjp669wvgq4-dear-imgui-plan-to-nix-pkgs/.plan.nix/dear-imgui.nix:48:126:
The Nixpkgs package set does not contain the package: sdl2 (system dependency).
You may need to augment the system package mapping in haskell.nix so that it can be found
```

Reply:
on darwin, doesn't this condition say that `sdl2` should be the haskell package?

```
if os(windows) || os(darwin)
      extra-libraries:
        sdl2
```

Reply:
No, `extra-libraries: sdl2` means that for `cabal build` to work `-lsdl2` needs to work. So this is about the libsdl dependency, not a Haskell dependency. We need this because we'll compile `imgui/backends/imgui_impl_sdl.cpp`, and that needs `-lsdl2`.